homeassistant/components/plum_lightpad/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Config flow for Plum Lightpad."""
+import logging
+from typing import Any, Dict, Optional
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import ConfigType
+
+from .const import DOMAIN
+from .utils import load_plum
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PlumLightpadConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Plum Lightpad integration."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input: Optional[ConfigType]) -> Dict[str, Any]:
+        """Handle a flow initialized by the user or redirected to by import."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(CONF_USERNAME): str,
+                        vol.Required(CONF_PASSWORD): str,
+                    }
+                ),
+            )
+
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+
+        # load Plum just so we know username/password work
+        plum = await load_plum(username, password, self.hass)
+        plum.cleanup()
```

Comment:
Because that instance of Plum is short lived - it is only created on line 39 to test credentials provided by user in UI. On line 40 it is 'destroyed'. The code does not make assumptions what cleanup() will do, just assumes it needs to be called if the instance is no longer in use, like here. Makes sense?

Reply:
Calling `plum.cleanup()` here isn't necessary here.

Reply:
removed